### PR TITLE
How to use K8up

### LINKED ISSUE #99: How to use K8up
The page ["Differences to APPUiO Public"](https://docs.appuio.cloud/user/explanation/differences-to-public.html#_latest_k8up_version) only mentions that

> Thanks to the recent Kubernetes version we offer the latest available K8up version

but there's no mention of how to use it.

Add a short page in the documentation explaining how to use it, and pointing to the [K8up main documentation](https://k8up.io/k8up/2.1/index.html) for more information. Do not explain K8up extensively, just do the following:

* Create a small dummy deployment including a PVC featuring the proper labels
* Deploy a small Minio server
* Create a `Backup` object in YAML and apply it
* Watch the backup repository appear on Minio

### PR DESCRIPTION
**Pull request by akosma**
Fixes #99 .

## Checklist

- [x] Try to isolate changes into separate PRs (to build a better changelog).
- [x] Categorize the PR by setting a good title and adding one of the labels:
      `how-to`, `tutorial`, `reference`, `explanation`, `cicd`, `dependency`
      as they show up in the changelog
- [x] Link this PR to related issues if applicable.

### CODE REVIEW

File: docs/modules/ROOT/pages/how-to/use-k8up.adoc [hunk 1]
```diff
@@ -0,0 +1,351 @@
...
+Apply the deployment to your {product} project:
+
+[source,shell]
+--
+oc apply -f deployment.yaml
+--
+
+=== 3. Deploy Minio
+
+K8up is powered by the https://restic.net/[Restic] backup tool, and is thus able to store backups in any S3-compatible storage. For the purposes of this guide, you can quickly install a https://min.io/[MinIO] instance in your project.
```
Comment (tobru) [inline docs/modules/ROOT/pages/how-to/use-k8up.adoc:75]: Maybe add a note that for production it's better to use an S3 endpoint outside of the cluster.

**Review by tobru — APPROVED** [review]
LGTM, let's get some more feedback from @Kidswiss and @ccremer

File: docs/modules/ROOT/pages/how-to/use-k8up.adoc [hunk 2]
```diff
@@ -0,0 +1,353 @@
...
+metadata:
+  name: minio
+spec:
+  selector:
+    matchLabels:
+      app: minio
+  strategy:
+    type: Recreate
+  template:
+    metadata:
```
Comment (ccremer) [inline docs/modules/ROOT/pages/how-to/use-k8up.adoc:112]: Why not install minio with Helm? (https://github.com/minio/minio/tree/master/helm/minio) Saves a lot of manually maintaining these deployment manifests here.
We'd probably need a values.yaml file to configure things like Ingress and credentials

File: docs/modules/ROOT/pages/how-to/use-k8up.adoc [hunk 3]
```diff
@@ -0,0 +1,353 @@
...
+      volumes:
+      - name: data
+        persistentVolumeClaim:
+          claimName: minio-pvc
+      containers:
+      - name: minio
+        volumeMounts:
+        - name: data
+          mountPath: "/data"
+        image: minio/minio
```
Comment (ccremer) [inline docs/modules/ROOT/pages/how-to/use-k8up.adoc:125]: With the Helm chart we wouldn't need to care about such details ;)
```suggestion
        image: quay.io/minio/minio
```

File: docs/modules/ROOT/pages/how-to/use-k8up.adoc [hunk 4]
```diff
@@ -0,0 +1,353 @@
...
+      bucket: backups
+      accessKeyIDSecretRef:
+        name: minio-credentials
+        key: username
+      secretAccessKeySecretRef:
+        name: minio-credentials
+        key: password
+--
+<1> This `Secret` is used to store the password used by Restic to access the backup repository stored in MinIO. *This credential is only for demo purposes!*
+
```
Comment (ccremer) [inline docs/modules/ROOT/pages/how-to/use-k8up.adoc:273]: I guess we need to apply the manifests?
e.g. `oc apply -f backup.yaml`

File: docs/modules/ROOT/pages/how-to/use-k8up.adoc [hunk 5]
```diff
@@ -0,0 +1,353 @@
...
+[source,shell]
+--
+restic restore latest --target ./restored
+--
+
+The file `restored/[PROJECT_NAME]-fortune-container.txt` will contain a fortune message coming from the deployment.
+
+=== 6. Schedule regular backups
+
+A `Backup` object performs a one-off backup of your pods and PVCs. You can instead create a `Schedule` object to backup your system regularly:
```
Comment (ccremer) [inline docs/modules/ROOT/pages/how-to/use-k8up.adoc:319]:
```suggestion
A `Backup` object performs a one-off backup of your pods and PVCs.
You can instead create a `Schedule` object to backup your system regularly:
```

File: docs/modules/ROOT/pages/how-to/use-k8up.adoc [hunk 6]
```diff
@@ -0,0 +1,353 @@
...
+    failedJobsHistoryLimit: 2
+    successfulJobsHistoryLimit: 2
+--
+<1> This instruction will trigger a backup every 5 minutes.
+
+== Tips & Tricks
+
+Here go some ideas to use K8up efficiently:
+
+* K8up has plenty of options. Check the https://k8up.io/[documentation] to learn more.
```
Comment (ccremer) [inline docs/modules/ROOT/pages/how-to/use-k8up.adoc:352]:
```suggestion
* K8up has plenty of options.
  Check the https://k8up.io/[documentation] to learn more.
```

**Review by ccremer — CHANGES_REQUESTED** [review]

File: docs/modules/ROOT/pages/how-to/use-k8up.adoc (hunk 2 above)
Comment (akosma) [reply docs/modules/ROOT/pages/how-to/use-k8up.adoc:121]: Let's use this now and consider the Helm chart for a future version.

File: docs/modules/ROOT/pages/how-to/use-k8up.adoc [hunk 7]
```diff
@@ -0,0 +1,373 @@
...
+<3> These are the MinIO credentials.
+
+Source the file to load the variables in your environment:
+
+[source,shell]
+--
+source environment.sh
+--
+
+Use `restic` to watch the current snapshots:
```
Comment (Kidswiss) [inline docs/modules/ROOT/pages/how-to/use-k8up.adoc:319]:
```suggestion
Use `restic` to list the current snapshots:
```

File: docs/modules/ROOT/pages/how-to/use-k8up.adoc [hunk 8]
```diff
@@ -0,0 +1,373 @@
...
+
+[source,shell]
+--
+restic snapshots
+--
+
+You can now backup the latest snapshot locally into the `restored` folder:
```
Comment (Kidswiss) [inline docs/modules/ROOT/pages/how-to/use-k8up.adoc:326]: Maybe an example with a restore directly to a PVC would make more sense?

Comment (akosma) [reply docs/modules/ROOT/pages/how-to/use-k8up.adoc:326]: Well, there's the K8up website and its getting started guide for that. Just wanted to show that effectively, stuff is backed up.

File: docs/modules/ROOT/pages/how-to/use-k8up.adoc [hunk 9]
```diff
@@ -0,0 +1,373 @@
...
+. Create a new project called "[YOUR_USERNAME]-fortune-go"
++
+[source,shell]
+--
+oc new-project [YOUR_USERNAME]-fortune-go
+--
+
+=== 2. Deploy a simple app
+
+We're going to deploy a https://gitlab.com/vshn/applications/fortune-go[sample application written in Go] using the `deployment.yaml` file shown below. This deployment doesn't need neither a `Service` nor an `Ingress`:
```
Comment (ccremer) [inline docs/modules/ROOT/pages/how-to/use-k8up.adoc:38]:
```suggestion
We're going to deploy a https://gitlab.com/vshn/applications/fortune-go[sample application written in Go] using the `deployment.yaml` file shown below.
This deployment doesn't need neither a `Service` nor an `Ingress`:
```

**Review by ccremer — APPROVED** [review]

**Review by Kidswiss — APPROVED** [review]

**Pull request merged**